integrations/tensorflow/compiler/dialect/tf_tensorlist/conversion/convert_flow_to_hal.cc

```diff
@@ -16,21 +16,131 @@
+    return srcOp->emitOpError() << "unable to create adaptor for operand";
+  }
+  auto bufferView = operand->getBufferView();
+  if (!bufferView) {
+    return srcOp->emitOpError() << "unable to get buffer view for operand";
+  }
+
+  retValue = bufferView;
+  return success();
+}
+
+class ReserveOpConversion : public OpConversionPattern<tf_tensorlist::Reserve> {
+ public:
+  ReserveOpConversion(MLIRContext *ctx, TypeConverter &converter)
+      : OpConversionPattern(ctx) {}
+
+  LogicalResult matchAndRewrite(
+      tf_tensorlist::Reserve reserveOp, llvm::ArrayRef<Value> newOperands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto elementTy = reserveOp.element_type();
+    auto element_value = IREE::HAL::getElementTypeValue(elementTy).getValue();
+
+    Value operand0, operand1;
+    getBufferView(reserveOp, reserveOp.getOperand(0), newOperands[0], rewriter,
+                  operand0);
+    getBufferView(reserveOp, reserveOp.getOperand(1), newOperands[1], rewriter,
+                  operand1);
+
+    rewriter.replaceOpWithNewOp<IREE::TensorList::Reserve>(
+        reserveOp,
+        IREE::TensorList::TensorListType::get(reserveOp.getContext()), operand0,
+        operand1, rewriter.getI32IntegerAttr(element_value));
+    return success();
+  }
+};
+
+class ConcatOpConversion : public OpConversionPattern<tf_tensorlist::Concat> {
+ public:
+  ConcatOpConversion(MLIRContext *ctx, TypeConverter &converter)
+      : OpConversionPattern(ctx) {}
+
+  LogicalResult matchAndRewrite(
+      tf_tensorlist::Concat concatOp, llvm::ArrayRef<Value> newOperands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto device =
+        rewriter.createOrFold<IREE::HAL::ExSharedDeviceOp>(concatOp.getLoc());
+    auto allocator =
+        rewriter.create<IREE::HAL::DeviceAllocatorOp>(concatOp.getLoc(), device)
+            .getResult();
+
+    auto newConcatOp = rewriter.createOrFold<IREE::TensorList::Concat>(
+        concatOp.getLoc(),
+        IREE::HAL::BufferViewType::get(rewriter.getContext()), allocator,
+        newOperands[0]);
+
+    auto bufferOp = rewriter.createOrFold<IREE::HAL::BufferViewBufferOp>(
+        newConcatOp.getLoc(), newConcatOp);
+
+    rewriter.replaceOp(concatOp, bufferOp);
+    return success();
+  }
+};
+
+class StackOpConversion : public OpConversionPattern<tf_tensorlist::Stack> {
+ public:
+  StackOpConversion(MLIRContext *ctx, TypeConverter &converter)
+      : OpConversionPattern(ctx) {}
+
+  LogicalResult matchAndRewrite(
+      tf_tensorlist::Stack stackOp, llvm::ArrayRef<Value> newOperands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto device =
+        rewriter.createOrFold<IREE::HAL::ExSharedDeviceOp>(stackOp.getLoc());
+    auto allocator =
+        rewriter.create<IREE::HAL::DeviceAllocatorOp>(stackOp.getLoc(), device)
+            .getResult();
+
+    Value operand1;
+    getBufferView(stackOp, stackOp.getOperand(1), newOperands[1], rewriter,
```

Comment:
`Value operand1 = getBufferView(...)`? You don't seem to be checking the LogicalResult. You can just return nullptr  in case of failure (and actually check it)

Reply:
Done.